crates/driver/src/infra/tokens.rs

```diff
@@ -0,0 +1,88 @@
+use {
+    crate::{domain::eth, infra::Ethereum},
+    anyhow::Result,
+    contracts::ERC20,
+    ethcontract::dyns::DynWeb3,
+    std::{
+        collections::HashMap,
+        sync::{Arc, RwLock},
+    },
+};
+
+#[derive(Clone, Debug, Default)]
+pub struct Metadata {
+    pub decimals: u8,
+    pub symbol: String,
+}
```

Comment:
Might make sense to move this into the `domain` module next to the `auction::Tokens` type, as this is core to the domain logic (i.e. it gets included with the tokens in the augmented auction).